diff --git a/gdk-pixbuf-loader/io-svg.c b/gdk-pixbuf-loader/io-svg.c
index bcdfd0bb..3c31f757 100644
--- a/gdk-pixbuf-loader/io-svg.c
+++ b/gdk-pixbuf-loader/io-svg.c
@@ -26,7 +26,6 @@
 #include <stdlib.h>
 
 #include <rsvg.h>
-#include <gdk-pixbuf/gdk-pixbuf.h>
 
 #define N_(string) (string)
 #define _(string) (string)
@@ -43,8 +42,8 @@ typedef struct {
         gpointer                    user_data;
 } SvgContext;
 
-G_MODULE_EXPORT void fill_vtable (GdkPixbufModule *module);
-G_MODULE_EXPORT void fill_info (GdkPixbufFormat *info);
+// G_MODULE_EXPORT void fill_vtable (GdkPixbufModule *module);
+// G_MODULE_EXPORT void fill_info (GdkPixbufFormat *info);
 
 enum {
         ERROR_WRITING = 1,
@@ -173,16 +172,20 @@ gdk_pixbuf__svg_image_stop_load (gpointer data, GError **error)
         return result;
 }
 
-void
-fill_vtable (GdkPixbufModule *module)
+#ifndef INCLUDE_svg
+#define MODULE_ENTRY(function) G_MODULE_EXPORT void function
+#else
+#define MODULE_ENTRY(function) void _gdk_pixbuf__svg_ ## function
+#endif
+
+MODULE_ENTRY (fill_vtable) (GdkPixbufModule *module)
 {
         module->begin_load     = gdk_pixbuf__svg_image_begin_load;
         module->stop_load      = gdk_pixbuf__svg_image_stop_load;
         module->load_increment = gdk_pixbuf__svg_image_load_increment;
 }
 
-void
-fill_info (GdkPixbufFormat *info)
+MODULE_ENTRY (fill_info) (GdkPixbufFormat *info)
 {
         static const GdkPixbufModulePattern signature[] = {
                 {  " <svg",  "*    ", 100 },
diff --git a/rsvg.h b/rsvg.h
index e4350da7..1ee4c870 100644
--- a/rsvg.h
+++ b/rsvg.h
@@ -31,7 +31,80 @@
 #include <glib-object.h>
 #include <gio/gio.h>
 
+#ifndef INCLUDE_svg
 #include <gdk-pixbuf/gdk-pixbuf.h>
+#else
+// typedef enum {
+//     STORAGE_UNINITIALIZED,
+//     STORAGE_PIXELS,
+//     STORAGE_BYTES
+// } Storage;
+// typedef void (* GdkPixbufDestroyNotify) (guchar *pixels, gpointer data);
+// extern struct Pixels {
+//     guchar *pixels;
+//     GdkPixbufDestroyNotify destroy_fn;
+//     gpointer destroy_fn_data;
+// };
+// extern struct Bytes {
+//         GBytes *bytes;
+// };
+struct _GdkPixbuf; //{
+//     GObject parent_instance;
+// 	GdkColorspace colorspace;
+// 	int n_channels;
+// 	int bits_per_sample;
+// 	int width, height;
+// 	int rowstride;
+//     Storage storage;
+//     struct {
+//         Pixels pixels;
+//         Bytes bytes;
+//     } s;
+// 	guint has_alpha : 1;
+// };
+typedef struct _GdkPixbuf GdkPixbuf;
+struct _GdkPixbufLoader;
+typedef struct _GdkPixbufLoader GdkPixbufLoader;
+struct _GdkPixbufFormat {
+  gchar *name;
+  GdkPixbufModulePattern *signature;
+  gchar *domain;
+  gchar *description;
+  gchar **mime_types;
+  gchar **extensions;
+  guint32 flags;
+  gboolean disabled;
+  gchar *license;
+};
+typedef struct _GdkPixbufFormat GdkPixbufFormat;
+struct _GdkPixbufAnimation;
+typedef struct _GdkPixbufAnimation GdkPixbufAnimation;
+struct _GdkPixbufModulePattern;
+typedef struct _GdkPixbufModulePattern GdkPixbufModulePattern;
+extern GQuark gdk_pixbuf_error_quark(void);
+#define GDK_PIXBUF_ERROR gdk_pixbuf_error_quark ()
+#define GDK_PIXBUF_ERROR_FAILED 5
+#define GDK_COLORSPACE_RGB 0
+typedef void (* GdkPixbufModuleUpdatedFunc) (GdkPixbuf*, int, int, int, int, gpointer);
+typedef void (* GdkPixbufModulePreparedFunc) (GdkPixbuf, GdkPixbufAnimation *anim, gpointer);
+typedef void (* GdkPixbufModuleSizeFunc) (gint*, gint*, gpointer);
+// extern void GdkPixbufModuleUpdatedFunc(struct GdkPixbuf *, int, int, int, int, void *);
+// extern void GdkPixbufModulePreparedFunc(struct GdkPixbuf *, void *, void *);
+// extern void GdkPixbufModuleSizeFunc(int, int, void *);
+// // typedef bool gboolean;
+// // typedef void* gpointer;
+// // typedef char gchar;
+// // extern struct GError;
+// extern struct GdkPixbufModulePattern;
+// extern struct GdkPixbufModule;
+// extern struct GdkPixbufFormat;
+// typedef enum
+// {
+//   GDK_PIXBUF_FORMAT_WRITABLE = 1 << 0,
+//   GDK_PIXBUF_FORMAT_SCALABLE = 1 << 1,
+//   GDK_PIXBUF_FORMAT_THREADSAFE = 1 << 2
+// } GdkPixbufFormatFlags;
+#endif
 
 G_BEGIN_DECLS
 
