diff --git a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
index e1df5900f..ffe2873bb 100644
--- a/gdk-pixbuf/gdk-pixbuf-io.c
+++ b/gdk-pixbuf/gdk-pixbuf-io.c
@@ -47,6 +47,12 @@
 #include <mach-o/dyld.h>
 #endif
 
+// To avoid circular dependency we have to make a forward declaration of the required functions here
+#ifdef INCLUDE_svg
+extern void _gdk_pixbuf__svg_fill_info (GdkPixbufFormat *info);
+extern void _gdk_pixbuf__svg_fill_vtable (GdkPixbufModule *module);
+#endif
+
 /**
  * GdkPixbufModule:
  * @module_name: the name of the module, usually the same as the
@@ -659,6 +665,9 @@ gdk_pixbuf_io_init_builtin (void)
         /* Except the gdip-png loader which normally isn't built at all even */
         load_one_builtin_module (png);
 #endif
+#ifdef INCLUDE_svg
+        load_one_builtin_module (svg);
+#endif
 
 #undef load_one_builtin_module
 }
@@ -695,6 +704,7 @@ module (xbm);
 module (tga);
 module (icns);
 module (qtif);
+module (svg);
 module (gdip_ico);
 module (gdip_wmf);
 module (gdip_emf);
@@ -777,6 +787,9 @@ gdk_pixbuf_load_module_unlocked (GdkPixbufModule *image_module,
 #ifdef INCLUDE_qtif
         try_module (qtif,qtif);
 #endif
+#ifdef INCLUDE_svg
+        try_module (svg,svg);
+#endif
 
 #undef try_module
         
diff --git a/gdk-pixbuf/meson.build b/gdk-pixbuf/meson.build
index 570625bfe..0bc5ba4e2 100644
--- a/gdk-pixbuf/meson.build
+++ b/gdk-pixbuf/meson.build
@@ -327,11 +327,16 @@ if native_windows_loaders
   endif
 endif
 
+tools = get_option('tools')
+if tools
 gdkpixbuf_bin = [
   [ 'gdk-pixbuf-csource' ],
   [ 'gdk-pixbuf-pixdata' ],
   [ 'gdk-pixbuf-query-loaders', [ 'queryloaders.c' ] ],
 ]
+else
+gdkpixbuf_bin = []
+endif
 
 foreach bin: gdkpixbuf_bin
   bin_name = bin[0]
diff --git a/meson.build b/meson.build
index 3eb3fcc15..b11331239 100644
--- a/meson.build
+++ b/meson.build
@@ -280,6 +280,12 @@ if not gif_opt.disabled()
   enabled_loaders += 'gif'
 endif
 
+svg_opt = get_option('svg')
+if not svg_opt.disabled()
+  enabled_loaders += 'svg'
+  add_project_arguments([ '-DINCLUDE_svg' ], language: 'c')
+endif
+
 others_opt = get_option('others')
 if not others_opt.disabled()
   # Keep sorted alphabetically
diff --git a/meson_options.txt b/meson_options.txt
index 74830154a..e84ba78b8 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -18,6 +18,10 @@ option('others',
        description: 'Enable other loaders, which are weakly maintained',
        type: 'feature',
        value: 'disabled')
+option('svg',
+       description: 'Enable SVG loader (requires manual librsvg linking)',
+       type: 'feature',
+       value: 'disabled')
 option('builtin_loaders',
        description: 'Comma-separated list of loaders to build into gdk-pixbuf',
        type: 'array',
@@ -61,4 +65,7 @@ option('gio_sniffing',
        description: 'Perform file type detection using GIO (Unused on MacOS and Windows)',
        type: 'boolean',
        value: true)
-
+option('tools',
+       description: 'Build the tools',
+       type: 'boolean',
+       value: true)
