diff --git a/gdk-pixbuf/gdk-pixbuf-io.c b/gdk-pixbuf/gdk-pixbuf-io.c
index e207d8143..23ddf950e 100644
--- a/gdk-pixbuf/gdk-pixbuf-io.c
+++ b/gdk-pixbuf/gdk-pixbuf-io.c
@@ -660,6 +660,8 @@ gdk_pixbuf_io_init_builtin (void)
         load_one_builtin_module (png);
 #endif
 
+load_one_builtin_module (svg);
+
 #undef load_one_builtin_module
 }
 
@@ -695,6 +697,7 @@ module (xbm);
 module (tga);
 module (icns);
 module (qtif);
+module (svg);
 module (gdip_ico);
 module (gdip_wmf);
 module (gdip_emf);
@@ -777,6 +780,7 @@ gdk_pixbuf_load_module_unlocked (GdkPixbufModule *image_module,
 #ifdef INCLUDE_qtif
         try_module (qtif,qtif);
 #endif
+try_module (svg,svg);
 
 #undef try_module
         
